cmd/cstor-volume-mgmt/controller/volume-controller/handler.go

```diff
@@ -0,0 +1,150 @@
+package volumecontroller
+
+import (
+	"fmt"
+	"os"
+	"reflect"
+
+	"github.com/golang/glog"
+	"github.com/openebs/maya/cmd/cstor-volume-mgmt/controller/common"
+	"github.com/openebs/maya/cmd/cstor-volume-mgmt/volume"
+	apis "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/tools/cache"
+)
+
+// syncHandler compares the actual state with the desired, and attempts to
+// converge the two. It then updates the Status block of the cStorVolumeUpdated resource
+// with the current status of the resource.
+func (c *CStorVolumeController) syncHandler(key string, operation common.QueueOperation) error {
+	glog.Infof("at sync handler")
+	cStorVolumeGot, err := c.getVolumeResource(key)
+	if err != nil {
+		return err
+	}
+	status, err := c.cStorVolumeEventHandler(operation, cStorVolumeGot)
+	if status == common.CVStatusIgnore {
+		return nil
+	}
+	cStorVolumeGot.Status.Phase = string(status)
+	if err != nil {
+		_, err := c.clientset.OpenebsV1alpha1().CStorVolumes().Update(cStorVolumeGot)
+		if err != nil {
+			return err
+		}
+		return err
+	}
+	_, err = c.clientset.OpenebsV1alpha1().CStorVolumes().Update(cStorVolumeGot)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// cStorVolumeEventHandler is to handle cstor volume related events.
+func (c *CStorVolumeController) cStorVolumeEventHandler(operation common.QueueOperation, cStorVolumeGot *apis.CStorVolume) (common.CStorVolumeStatus, error) {
+	// volume.FileOperatorVar = util.RealFileOperator{}
+	// volume.UnixSockVar = util.RealUnixSock{}
+	glog.Infof("%v event received for volume : %v ", operation, cStorVolumeGot.Spec.VolumeName)
+	switch operation {
+	case common.QOpAdd:
+		// CheckValidVolume is to check if volume attributes are correct.
+		err := volume.CheckValidVolume(cStorVolumeGot)
+		if err != nil {
+			return common.CVStatusOffline, err
+		}
+
+		err = volume.CreateVolume(cStorVolumeGot)
+		if err != nil {
+			return common.CVStatusFailed, err
+		}
+		break
+
+	case common.QOpModify:
+		err := volume.CheckValidVolume(cStorVolumeGot)
+		if err != nil {
+			return common.CVStatusInvalid, err
+		}
+
+		err = volume.CreateVolume(cStorVolumeGot)
+		if err != nil {
+			return common.CVStatusFailed, err
+		}
+		break
+
+	case common.QOpDestroy:
+		return common.CVStatusIgnore, nil
+	}
+
+	return common.CVStatusIgnore, nil
+}
+
+// enqueueCstorVolume takes a CStorVolume resource and converts it into a namespace/name
+// string which is then put onto the work queue. This method should *not* be
+// passed resources of any type other than CStorVolumes.
+func (c *CStorVolumeController) enqueueCStorVolume(obj *apis.CStorVolume, q common.QueueLoad) {
+	var key string
+	var err error
+	if key, err = cache.MetaNamespaceKeyFunc(obj); err != nil {
+		runtime.HandleError(err)
```

Comment:
Does this operation log the error message? If not, it is better to log the same.